### ISSUE TITLE
The Community Administrator should not view all communities in the create collection section

### ISSUE THREAD

**Issue by francescopioscognamiglio** [DSpace/DSpace#8918]
**Describe the bug**
The Create collection section displays all collections to the Community Administrator, so even collections that are not within a community for which the user is a community administrator.

**To Reproduce**
Directly on the demo instance https://demo7.dspace.org/home
1. Log in as Community Administrator
2. Go to the "New" > "Collection" section
3. click any collection
4. fill the metadata
5. clicking on "Save" an "Access is denied" error is shown

Video dimonstrating the issue:
[screencast-demo7.dspace.org-2023.06.26-09_49_04.webm](https://github.com/DSpace/DSpace/assets/36161521/d86f8fc5-b5e6-4d79-a722-e2d62fb898b0)

**Expected behavior**
The list of collections should be filtered, so only collections within a community for which the current user is the community administrator should be shown.

**Comment by francescopioscognamiglio** [comment]
It seems to be related to https://github.com/DSpace/dspace-angular/issues/1331

**Comment by tdonohue** [comment]
Overlaps with #8916 and #8917 which are very similar issues (just in different UI screens).  Likely all three of these should be claimed together.   It is also similar to https://github.com/DSpace/dspace-angular/issues/1331

**Comment by PitbaranK** [comment]
Hi Tim,
I'm Pitbaran connecting you on behalf of Virsoftech Pvt. Ltd. We are interested in contributing to the DSpace open source repository. Specifically, we would like to claim this issue #8918 [Including #8916 & #8917].

Could you please guide us on the next steps and let us know if you require any additional details from our side to proceed.

**Comment by tdonohue** [comment]
Hi @PitbaranK :  Thanks for volunteering!  I'll assign this ticket to you just to claim it for Virsoftech.  But, anyone in your organization can complete the work.

Keep in mind, for work to be accepted, it must follow our guidelines for code contributing pull requests (PRs):
* Code guidelines for backend (DSpace/DSpace): https://github.com/DSpace/DSpace/blob/main/CONTRIBUTING.md
* Code guidelines for frontend (DSpace/dspace-angular): https://github.com/DSpace/dspace-angular/blob/main/CONTRIBUTING.md

Reviewers will check if PRs align with these guidelines. PRs which follow these guidelines are easier to get merged.  PRs which do not follow those guidelines may end up rejected or delayed.  If you have any questions, please feel free to ask!

If you run into issues or need more feedback as you begin development, you are welcome to create a draft PR and ask for early feedback on it.  This may be useful if you are not sure if a solution/fix is entirely correct or could be improved...in that scenario, a draft PR might provide other developers a chance to give you hints/tips on how to improve the PR.

**Comment by PitbaranK** [comment]
Hi @tdonohue,
Currently, our authorization layer is not directly integrated into SOLR. Instead, we utilize the **getSearchObjects** method from **DiscoveryRestRepository** to obtain pageable results, which we then filter based on the current user's authorization. This approach requires modifying the pageable results after retrieving them.

Can you please assist and provide your feedback on this approach? Or if there is any alternate way to directly filter out the **IndexableObjects** from SOLR based on user authorization please assist. 

Please let us know if there is any other query from your side.

**Comment by tdonohue** [comment]
@PitbaranK : Based on your comments, I've realized this ticket **might be a duplicate of** (or directly related to) https://github.com/DSpace/dspace-angular/issues/1331 (which Atmire has already begun work on)

@artlowel : Are you aware of whether this issue is already being fixed by Atmire?   I don't want there to be duplicative work in this area, and this ticket and https://github.com/DSpace/dspace-angular/issues/1331 (assigned to you) might be describing the same essential problem

If these tickets are *NOT* duplicates, then it might be that the solution to this ticket should be _similar to_ the solution created for https://github.com/DSpace/dspace-angular/issues/1331

**Comment by artlowel** [comment]
Yes. This looks to be a subset of the work @ybnd and @KoenP were doing in https://github.com/DSpace/DSpace/pull/8616

If I recall correctly, it was put on hold because there were performance issues keeping solr up to date after an edit, and there was no real consensus on how to proceed

Nobody is actively working on it at the moment

**Comment by PitbaranK** [comment]
Hi @tdonohue,
We have raised PR for the bug #8918 (Including #8916 and #8917) by @im-shubham-vish and are awaiting reviewer feedback from long time. Please let us know the current status and also if there's anything we may have missed in the submission.

**Comment by tdonohue** [comment]
Backported to `dspace-7_x` in DSpace/dspace-angular#5123 and fixed in 7.6.7
